### PR TITLE
minor changes in mod(a, b)

### LINKED ISSUE #26016: i think mod(a,b) are considering only integers solutions
Hello guys, i have a simplification of my code which is:

```
import sympy as sp
import itertools

Zp = range(3)
Zq = range(3) 
sextetos = itertools.product(Zp, Zp, Zp, Zq, Zq, Zq)


x, y, z, w = sp.symbols('x y z w', real=True)
system = []
#it will test the generic form: B(a,b) = x*a1*b1+y*a1*b2+z*a2*b1+w*a2*b2, for all pairs a= (a1,a2), b=(b1,b2) e c=(c1,c2)
#both mapping axioms, and return the set of conditions that x, y, z e w must satisfy.

for a1, b1, c1, a2, b2, c2 in sextetos:
 d1 = (b1 + c1) % p
 d2 = (b2 + c2) % q   
 if (a1, b1, c1, a2, b2, c2) != (0, b1, c1, 0, b2, c2):
    print(a1,b1,c1,a2,b2,c2)
    eq1 = ((x * (a1 * b1+b1 * a1)+y * (a1 * b2+b1 * a2)+z * (a2*b1+b2 * a1)+w * (a2 * b2+a2 * b2))) % 2
    system.append(eq1)
    eq2 = (x * (a1 * ((b1+c1)%p))+y * (a1 * ((b2+c2)%q))+z * (a2 * ((b1+c1)%p))+w * (a2*((b2+c2)%q)) - (x * a1 * (b1+c1)+y * a1 * (b2+c2)+z * a2*(b1+c1)+w * a2 * (b2+c2))) % 2
    eq2 = (x * a1 * (sp.Mod(b1+c1,p)-(b1+c1))+y * a1 * (sp.Mod(b2+c2,q)-(b2+c2))+z * a2 * (sp.Mod(b1+c1,p)-(b1+c1))+w * a2 * (sp.Mod(b2+c2,q)-(b2+c2))) % 2
    print(eq1)
    print(eq2)
    input()
vars = [x, y, z, w]
print(sp.nonlinsolve(system, vars)) 
```

I was feeling something was a litte bit wrong with this code, because the conditionset it returns has some rules that i know that doesnt exist, like for the sexteto (a1,b1,c1,a2,b2,c2) = (0,0,0,1,1,2), eq2 should be w * 1 * (-3) % 2. Or, with sp tools: Eq(3*Mod(w,2),0), considering the possibility of w = 2/3 (example). However, the code returns me just Eq(Mod(w,2),0), and i just cant understand why...

Important that: the error is before the nonlinsolve function. If u just put a print(eq2) with the equations.append(eq2), it will return (Eq(Mod(w,2),0). 

If someone have an ideia of what is happening, please give me a hand. 

Merry Christmas, gifhubbers :D

### PR DESCRIPTION
**Pull request by harshkasat**
#### References to other Issues or PRs

Fixes #26016

#### Release Notes

* core
  * minor changes in mod as mentioned in [commnet](https://github.com/sympy/sympy/issues/26016#issuecomment-1873374064)

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Tests should be added for what is fixed.

File: sympy/core/tests/test_args.py [hunk 1]
```diff
@@ -682,6 +682,7 @@ def test_sympy__core__function__WildFunction():
 def test_sympy__core__mod__Mod():
     from sympy.core.mod import Mod
     assert _test_args(Mod(x, 2))
+    assert _test_args(Mod(3*x, 2))
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_args.py:685]: This test would pass without the changes here. This does not really test anything useful and should be removed.

Tests should be added in `sympy/core/tests/test_arit.py`.

File: sympy/core/tests/test_arit.py [hunk 2]
```diff
@@ -1996,6 +1996,8 @@ def test_Mod():
     from sympy.abc import phi
     assert Mod(4.0*Mod(phi, 1) , 2) == 2.0*(Mod(2*(Mod(phi, 1)), 1))
 
+    assert Mod(3*x, 2) == Mod(3*x, 2)
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_arit.py:1999]: This test does not do anything: it would already pass without the changes here.

Comment (oscarbenjamin) [reply sympy/core/tests/test_arit.py:1999]: The test should be
```
assert unchanged(Mod, 3*x, 2)
```

Comment (oscarbenjamin) [reply sympy/core/tests/test_arit.py:1999]: There should also be a test for what happens with a symbol that has `integer=True` set (if there is not one already).

Comment (harshkasat) [reply sympy/core/tests/test_arit.py:1999]:
```python
x = symbols('x', integer=True)
unchanged(Mod, 3*x, 2) 
``` 
Give output False, as [mentioned ](https://github.com/sympy/sympy/issues/26016#issuecomment-1873350941)here 
> This simplification assumes that x is an integer which is not necessarily the case if no assumptions are set on x:

Comment (oscarbenjamin) [reply sympy/core/tests/test_arit.py:1999]: The test should be:
```
xi = symbols('x', integer=True)
assert unchanged(Mod, xi, 2) 
assert Mod(3*xi, 2) == Mod(xi, 2)
```

**Pull request merged**